File: hello.py
import os 
import pandas as pd
from time import sleep
import requests
import logging
import validators
from datetime import datetime

qtde_monitoramentos = 3
delayMonitoramento = 5

def exibeIntroducao():
    nomePessoa = "Jackson"
    versao = 1.1

    print("Olá Sr.", nomePessoa, ".")
    print("Este programa esta na versão: ", versao)

def exibeMenu():
    print("1- Iniciar Monitoramento")
    print("2- Exibir os Logs")
    print("0- Sair do Programa")

def lerComando():
    escolha = int(input("Escolha uma opção:")) 
    print("O Comando escolhido foi:", escolha)

    return escolha

def leSitesDoArquivo():
    lines = []
    file = 'listasites.txt'
    with open(file) as f:
        [lines.append(line.rstrip('\n')) for line in f.readlines()] 
    
    return lines

def registraLog(site, status):
    
    logging.basicConfig(filename='logHello_Py.log', format='%(asctime)s - %(name)s - %(levelname)s - %(message)s', level = logging.INFO)

    if status:
        logging.info(f"Testando {site} - Online {status}")
    else:
        logging.error(f"Testando {site} - Online {status}")

def testaSiteOnline(site_para_teste):
    
    if validators.url(site_para_teste):
        res=requests.get(site_para_teste)
        if res.status_code == 200:
            print("site:", site_para_teste, "foi carregado com sucesso !!")
            registraLog(site_para_teste, True)
        else:
            print("O Site:", site_para_teste, "esta com problemas. Status code:", res.status_code)
            registraLog(site_para_teste, False)
    else:
        logging.error(f"Testando: {site_para_teste} - Status: URL é Inválida")

def iniciarMonitoramento():

    print("Monitorando.....")
    sites = list(leSitesDoArquivo())

    for i in range(qtde_monitoramentos):
        for site in sites:
            print("Testando Site", i, ":", site)
            testaSiteOnline(site)

        sleep(delayMonitoramento)
        print("")

    print("")

def imprimirLogs():
    try:
        with open('logHello_Py.log', 'r') as f:
            print(f.read())
    except IOError:
        print(u'Arquivo de log não encontrado!')

if __name__ == '__main__':
    
    exibeIntroducao()

    while True:
        exibeMenu()

        match lerComando():
            case 1:
                iniciarMonitoramento()
            case 2:
                imprimirLogs()
            case 0:
                print("Saindo...tchau...Pythoniacos")
                os._exit(0)
            case _:
                print("Opção escolhida não é valida.")
                os._exit(-1)
	

File: hello.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"io/ioutil"
	"net/http"
	"os"
	"strconv"
	"strings"
	"time"
)

const monitoramentos = 3
const delayMonitoramento = 5 * time.Second

func main() {

	exibeIntroducao()

	for {
		exibeMenu()
		//comando := lerComando()

		switch lerComando() {
		case 1:
			iniciarMonitoramento()
			break /* não é necessário, por default ele vai dar um break caso ache a opção desejada*/
		case 2:
			imprimirLogs()
		case 0:
			fmt.Println("Saindo...tchau...GoLangers")
			os.Exit(0)
		default:
			fmt.Println("Opção escolhida não é valida.")
			os.Exit(-1)
		}
	}
}

func exibeIntroducao() {
	nomePessoa := "Jackson"
	versao := float32(1.1) /* se não informar o tipo, o Go irá criar como float64 - sempre*/

	fmt.Println("Olá Sr.", nomePessoa, ".")
	fmt.Println("Este programa esta na versão: ", versao)
}

func exibeMenu() {
	fmt.Println("1- Iniciar Monitoramento")
	fmt.Println("2- Exibir os Logs")
	fmt.Println("0- Sair do Programa")
}

func lerComando() int {
	var escolha int
	fmt.Scan(&escolha) /* & indica o ponteiro da variavel escolha, para que seja preenchido com o que o usuario informar */
	fmt.Println("O Comando escolhido foi:", escolha)

	return escolha
}

func iniciarMonitoramento() {

	fmt.Println("Monitorando.....")
	sites := leSitesDoArquivo()

	for i := 0; i < monitoramentos; i++ {
		for indice, site := range sites {
			fmt.Println("Testando Site", indice, ":", site)
			testaSiteOnline(site)
		}
		time.Sleep(delayMonitoramento)
		fmt.Println("")
	}

	fmt.Println("")
}

func testaSiteOnline(site string) {

	resp, err := http.Get(site)

	if err != nil {
		fmt.Println("Ocorreu um Erro : ", err)
	}

	if resp.StatusCode == 200 {
		fmt.Println("site:", site, "foi carregado com sucesso !!")
		registraLog(site, true)
	} else {
		fmt.Println("O Site:", site, "esta com problemas. Status code:", resp.StatusCode)
		registraLog(site, false)
	}
}

func leSitesDoArquivo() []string {

	var sites []string
	arquivo, err := os.Open("listasites.txt")

	if err != nil {
		fmt.Println("Ocorreu um erro: ", err)
	}

	leitor := bufio.NewReader(arquivo)

	for {
		linha, err := leitor.ReadString('\n')
		linha = strings.TrimSpace(linha)

		sites = append(sites, linha)

		if err == io.EOF {
			break
		}
	}

	arquivo.Close()
	return sites
}

func registraLog(site string, status bool) {

	arquivoLog, err := os.OpenFile("logHello_Go.log", os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
	if err != nil {
		fmt.Println("Erro ao abrir o arquivo:", err)
	}

	horalog := time.Now().Format("02/01/2006 15:04:05")
	arquivoLog.WriteString(horalog + " - " + site + " - Online: " + strconv.FormatBool(status) + "\n")
	arquivoLog.Close()
}

func imprimirLogs() {

	arquivo, err := ioutil.ReadFile("logHello_Go.log") /* ler tudo de uma vez e imprimir */
	if err != nil {
		fmt.Println("Ocorreu um erro: ", err)
	}

	fmt.Println(string(arquivo))
}
